cpp/extract_normdata_translations.cc

```diff
@@ -0,0 +1,225 @@
+                ++it;
+            }
+            else if (*it == "IxTheo_fra") {
+                term_to_translation_map[fr].emplace(StringUtil::Join(german_term, ' '), *(it+1));
+                ++it;
+            }
+
+            else if (*it == "lcsh") {
+                term_to_translation_map[en].emplace(StringUtil::Join(german_term, ' '), *(it+1));
+                ++it;
+            }
+            else if (*it == "ram") {
+                term_to_translation_map[fr].emplace(StringUtil::Join(german_term, ' '), *(it+1));
+                ++it;
+            }
+            
+        }
+       
+    }
+
+    count++;
+
+}
+
+
+std::unique_ptr<File> OpenInputFile(const std::string &filename) {
+    std::string mode("r");
+    mode += MediaTypeUtil::GetFileMediaType(filename) == "application/lz4" ? "u" : "m";
+    std::unique_ptr<File> file(new File(filename, mode));
+    if (file == nullptr)
+        Error("can't open \"" + filename + "\" for reading!");
+
+    return file;
+}
+
+
+
+int main(int argc, char **argv) {
+    progname = argv[0];
+
+    if (argc != 3)
+        Usage();
+
+    const std::string norm_data_marc_input_filename(argv[1]);
+    std::unique_ptr<File> norm_data_marc_input(OpenInputFile(norm_data_marc_input_filename));
+
+    const std::string extracted_translations_filename(argv[2]);
+    if (unlikely(norm_data_marc_input_filename == extracted_translations_filename))
+        Error("Norm data input file name equals output file name!");
+
+    std::string output_mode("w");
+    if (norm_data_marc_input->isCompressingOrUncompressing())
+        output_mode += 'c';
+    
+    // Create a file for each language
+    std::vector<std::string> output_file_components;
+    if (unlikely(StringUtil::Split(extracted_translations_filename, ".", &output_file_components) < 1)) 
+        Error("extracted_translations_filename " + extracted_translations_filename + " is not valid");
+   
+    std::vector<std::string> languages_to_create;
+    if (unlikely(StringUtil::Split(languages_to_create_str, ',', &languages_to_create) < 1))
+        Error("String " + languages_to_create_str + " is not valid");
+    
+    File* lang_file[NUMBER_OF_LANGUAGES];
+    unsigned i(0);
+  
+    // Derive output components from given input filename
+    std::string extension = (output_file_components.size() > 1) ? output_file_components.back() : "";
+    std::string basename;
+    if (extension != "")
+        output_file_components.pop_back();
+    basename = StringUtil::Join(output_file_components, ".");
+      
+    // Assemble output filename
+    for (auto lang : languages_to_create) {
+         lang = StringUtil::Trim(lang);
+
+         std::string lang_file_name_str = 
+             (extension != "") ? basename + "_" + lang + "." + extension : basename + "_" + lang;
+
+         lang_file[i++] = new File(lang_file_name_str, output_mode);
+         if (lang_file[i] == NULL)
```

Comment:
1. Please use the safter "nullptr" and not "NULL", if that's what you need.
2. This code does not what you think it does.  (See an earlier comment.)